bulk_indexer.go

```diff
@@ -232,9 +232,10 @@ func (b *bulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 	if b.maxDocumentRetry > 0 {
 		if cap(b.copyBuf) < b.buf.Len() {
 			b.copyBuf = slices.Grow(b.copyBuf, b.buf.Len()-cap(b.copyBuf))
-			b.copyBuf = b.copyBuf[:cap(b.copyBuf)]
+			b.copyBuf = b.copyBuf[:b.buf.Len()]
 		}
-		copy(b.copyBuf, b.buf.Bytes())
+		n := copy(b.copyBuf, b.buf.Bytes())
+		b.copyBuf = b.copyBuf[:n]
```

Comment:
Prior to this PR there was no error handling of the gzip reader.
I fixed that below, but that required resizing down `copyBuf` here when necessary - otherwise there might be unrelated data from previous flushes causing failed reads.